tests/test_detect.py

```diff
@@ -0,0 +1,212 @@
+# This file is part of lsst.scarlet.lite.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import os
+
+import numpy as np
+from lsst.scarlet.lite import Box, Image
+from lsst.scarlet.lite.detect import (
+    bounds_to_bbox,
+    get_detect_wavelets,
+    get_wavelets,
+    scarlet_footprints_to_image,
+)
+from lsst.scarlet.lite.detect_pybind11 import (
+    Footprint,
+    Peak,
+    get_connected_multipeak,
+    get_connected_pixels,
+    get_footprints,
+)
+from lsst.scarlet.lite.utils import integrated_circular_gaussian
+from numpy.testing import assert_array_equal
+from utils import ScarletTestCase
+
+
+class TestDetect(ScarletTestCase):
+    def setUp(self):
+        centers = (
+            (17, 9),
+            (27, 14),
+            (41, 25),
+            (10, 42),
+        )
+        sigmas = (1.0, 0.95, 0.9, 1.5)
+
+        sources = []
+        for sigma, center in zip(sigmas, centers):
+            yx0 = center[0] - 7, center[1] - 7
+            source = Image(integrated_circular_gaussian(sigma=sigma), yx0=yx0)
+            sources.append(source)
+
+        image = Image.from_box(Box((51, 51)))
+        for source in sources:
+            image += source
+        image.data[30:32, 40] = 0.5
+
+        self.image = image
+        self.centers = centers
+        self.sources = sources
+
+    def test_connected(self):
+        image = self.image.copy()
+
+        # Check that the first 3 footprints are all connected
+        # with thresholding at zero
+        truth = self.sources[0] + self.sources[1] + self.sources[2]
+        bbox = truth.bbox
+        truth = truth.data > 0
+
+        unchecked = np.ones(self.image.shape, dtype=bool)
+        footprint = np.zeros(self.image.shape, dtype=bool)
+        y, x = self.centers[0]
+        get_connected_pixels(
```

Comment:
I intentionally didn't so that I can keep track of changes to the API. In a way it's there to make it a pain to change the API and discourage me (or other people) from doing that.